packages/vue-app/template/observer.js

```diff
@@ -0,0 +1,49 @@
+// Inspired by: https://github.com/nuxt/image/blob/dev/src/runtime/utils/observer.ts
+
+let observerInstance
+let observerIdCtr = 1
+const OBSERVER_ID_KEY = '__nuxt_prefetch_observer__'
+
+export function get () {
+  if (typeof IntersectionObserver === 'undefined') {
+    return false
+  }
+
+  if (observerInstance) {
+    return observerInstance
+  }
+
+  const onIntersect = (entries) => {
+    entries.forEach(({ intersectionRatio, target: link }) => {
+      if (intersectionRatio <= 0 || !link.__prefetch) {
+        return
+      }
+      link.__prefetch()
+      remove(link) // Prefetching multiple times is not necessary
+    })
+  }
+  const intersectObserver = new IntersectionObserver(onIntersect)
+
+  const remove = (el) => {
+    if (!el[OBSERVER_ID_KEY]) { return }
+    delete el[OBSERVER_ID_KEY]
+    intersectObserver.unobserve(el)
+  }
+
+  const add = (el) => {
+    el[OBSERVER_ID_KEY] = el[OBSERVER_ID_KEY] || ++observerIdCtr
+    intersectObserver.observe(el)
+  }
+
+  observerInstance = { supported: true, add, remove }
+  return observerInstance
+}
+
+export function use (el) {
+  const observer = get()
+  observer.add(el)
+  if (!observer) {
+    return () => {}
```

Comment:
We may return a falsy value or in case of having callback for useObserver, immediately calling otherwise util consumer (nuxt-link) needs to again check `window.IntersectionObserver`

Reply:
I agree, there should be a nicer solution for it so the check only happens once.

Reply:
(updated)